src/backend/commands/tablecmds.c

```diff
@@ -13758,6 +13758,24 @@ ATPExecPartAdd(AlteredTableInfo *tab,
 					 errhint("use a named partition"),
 							   errOmitLocation(true)));
 
+	PartitionElem *pElem = (PartitionElem *) pc2->arg1;
+	Node *pStoreAttr = pElem->storeAttr;
+	if (pStoreAttr && ((AlterPartitionCmd *)pStoreAttr)->arg1)
+	{
+		List *pWithList = (List *)(((AlterPartitionCmd *)pStoreAttr)->arg1);
+		GpPolicy *parentPolicy = GpPolicyFetch(CurrentMemoryContext, RelationGetRelid(rel));
+		int bucketnum = parentPolicy->bucketnum;
+		int child_bucketnum = GetRelOpt_bucket_num_fromOptions(pWithList, bucketnum);
+
+		if (child_bucketnum != bucketnum)
+			ereport(ERROR,
+					(errcode(ERRCODE_GP_FEATURE_NOT_SUPPORTED),
+							errmsg("distribution policy for partition%s "
```

Comment:
Also, why this comment is not the same as the `analyze.c` below? E.g. wrap the `%s` in `\"%s\"`?